### ISSUE TITLE
Platform warning (linux/amd64) on raspberry pi

### ISSUE THREAD

**Issue by degreal** [packom/mbus-httpd#7]
- Raspberry PI Zero W
- M-Bus Master HAT DS

When following instructions and running command:
```
docker run --name mbus \
           -d -p 8080:8080 \
           -e RUST_LOG=INFO \
           --device /dev/ttyAMA0:/dev/ttyAMA0 \
           -v /proc/device-tree/hat:/proc/device-tree/hat \
           --privileged \
           packom/mbus-release
```
I get this warning:
`WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v6) and no specific platform was requested`

Is this something to worry about?

**Comment by piersfinlayson** [comment]
Yes, this probably is something that will cause a problem.  The error means that you're running on a different architecture platform (armv6 - the pi zero) to the platform the container image supports (x86-64/amd64).  I would expect that you'd hit a fatal error immediately running this container - either a segmentation fault or exec error.

The container image packom/mbus-release is actually a manifes that contains links to multiple different container images, for both x86 and arm architectures, although taking a quick look it may be that I haven't built one for armv6 (I do have images for other pis, which are armv7).  That surprises me though because I'm fairly sure I've tested on zeros!

On the other hand, it is also possible this is a benign warning, although I don't understand why you'd see it.

Does the container execute OK - do you see any logs if you run:
```
docker logs -f mbus
```

**Comment by degreal** [comment]
Looks like just docker images are incorrectly tagged. I get the same warning when using https://hub.docker.com/r/packom/mbus-release-arm/tags directly.

It works but I do some problems, but I will create new issue for that.

**Comment by nick-mitigo** [comment]
On RPI3 with Bullseye (2022-09-22) by default the steps are not working with docker...
this worked:
`sudo docker pull packom/mbus-release-arm:0.3.5`

this doesn't work:
`sudo docker run --name mbus -d -p 8080:8080 -e RUST_LOG=INFO --device /dev/ttyAMA0:/dev/ttyAMA0 -v /proc/device-tree/hat:/proc/device-tree/hat   --privileged packom/mbus-release-arm:0.3.5`

error: 
Error response from daemon: Cannot restart container 4ada: error while creating mount source path '/proc/device-tree/hat': mkdir /proc/device-tree/hat: operation not permitted

seems to work without the device tree parameter:
`sudo docker run --name mbus -d -p 8080:8080 -e RUST_LOG=INFO --device /dev/ttyAMA0:/dev/ttyAMA0 --privileged packom/mbus-release-arm:0.3.5`

with warning:
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested  2bfeea802e2f006395de1829099ba1596a781bdd16741cf0e697b66c3f6b578c

btw, wiringpi is also obsolete :)
